contracts/MultiRewards.sol

```diff
@@ -1,5 +1,22 @@
-pragma solidity 0.5.17;
+// pragma solidity 0.5.17;
 
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.12;
+pragma abicoder v2;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {IERC20Metadata} from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {Initializable} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import {PausableUpgradeable} from "@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol";
+
+import {IHypervisor} from "./interfaces/IHypervisor.sol";
```

Comment:
`IHypervisor` seems to be a partial interface for the Gamma vault at least based on it's usage:

```solidity
IHypervisor(stakingToken).getReward();
```

Presumably the `stakingToken` is the gamma vault token? If it is then the vault exposes a `getReward` function which likely collects and transfers any available rewards to this staker contract before the core logic of `stake`, `unstake`, `getReward`, etc to make a best effort at fair reward distribution.

Reply:
we will have similar function in our vaults, so this approach fits well with us